<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1168</width>
    <height>950</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>3D_Viewer_v1.0</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="qlview" name="openGLWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>900</width>
      <height>900</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="open_file">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>900</y>
      <width>261</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Open File</string>
    </property>
    <property name="flat">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QLabel" name="file_name">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>910</y>
      <width>631</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="save_jpeg">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>860</y>
      <width>116</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Pic</string>
    </property>
   </widget>
   <widget class="QPushButton" name="save_gif">
    <property name="geometry">
     <rect>
      <x>1040</x>
      <y>860</y>
      <width>116</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Gif</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_qty_vertice">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>930</y>
      <width>101</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Qty of vertices:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_qty_facet">
    <property name="geometry">
     <rect>
      <x>250</x>
      <y>930</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Qty of facets:</string>
    </property>
   </widget>
   <widget class="QSlider" name="x_move_slider">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>50</y>
      <width>161</width>
      <height>21</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::TicksBelow</enum>
    </property>
    <property name="tickInterval">
     <number>0</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_translating">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>10</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Translating</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_scale">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>320</y>
      <width>121</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Scale</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_x_move">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>40</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>X</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QSlider" name="scale_slider">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>340</y>
      <width>201</width>
      <height>31</height>
     </rect>
    </property>
    <property name="tracking">
     <bool>true</bool>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::TicksBelow</enum>
    </property>
   </widget>
   <widget class="QSlider" name="y_move_slider">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>90</y>
      <width>161</width>
      <height>21</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::TicksAbove</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_y_move">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>80</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QSlider" name="z_move_slider">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>130</y>
      <width>161</width>
      <height>21</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::TicksBelow</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_z_move">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>120</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QSlider" name="x_rotation_slider">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>210</y>
      <width>161</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <strikeout>false</strikeout>
      <kerning>true</kerning>
     </font>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::TicksBelow</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_y_rotation">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>240</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Y</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_x_rotation">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>200</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>X</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="label_rotation">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>160</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Rotation</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_z_rotation">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>280</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Z</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QSlider" name="z_rotation_slider">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>290</y>
      <width>161</width>
      <height>21</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::TicksBelow</enum>
    </property>
   </widget>
   <widget class="QSlider" name="y_rotation_slider">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>250</y>
      <width>161</width>
      <height>21</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::TicksBelow</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_type_projection">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>720</y>
      <width>201</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Type of projection:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_edges">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>420</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Edges:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_vertices">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>530</y>
      <width>81</width>
      <height>21</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>14</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Vertices:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_display_method_vertices">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>550</y>
      <width>101</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Display method:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="push_default_setting">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>770</y>
      <width>251</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Avenir</family>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Default setting</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>940</x>
      <y>830</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Save as:</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>590</y>
      <width>221</width>
      <height>32</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_6">
     <item>
      <widget class="QLabel" name="label_color_vertices">
       <property name="text">
        <string>Color</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="color_vertices_combo">
       <item>
        <property name="text">
         <string>white</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>black</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>blue</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>green</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>pink</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>purple</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>390</y>
      <width>141</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Avenir</family>
      <pointsize>18</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Setting</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>440</y>
      <width>221</width>
      <height>22</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLabel" name="label_edges_type">
       <property name="text">
        <string>Type:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="solid_button">
       <property name="text">
        <string>solid</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="dashed_button">
       <property name="text">
        <string>dashed</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>740</y>
      <width>221</width>
      <height>20</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QRadioButton" name="parallel_button">
       <property name="text">
        <string>parallel</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="central_button">
       <property name="text">
        <string>central</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>570</y>
      <width>221</width>
      <height>20</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QRadioButton" name="non_button">
       <property name="text">
        <string>None</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="circle_button">
       <property name="text">
        <string>Circle</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="square_button">
       <property name="text">
        <string>Square</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>460</y>
      <width>221</width>
      <height>32</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLabel" name="label_color_edges">
       <property name="text">
        <string>Color</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="color_edge_combo">
       <item>
        <property name="text">
         <string>white</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>blue</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>black</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>green</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>pink</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>purple</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>490</y>
      <width>221</width>
      <height>27</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <widget class="QLabel" name="label_thickness">
       <property name="text">
        <string>Thickness</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="thickness_edge_slider">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>620</y>
      <width>221</width>
      <height>27</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_7">
     <item>
      <widget class="QLabel" name="label_size_vertices">
       <property name="text">
        <string>Size</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="size_vertices_slider">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>670</y>
      <width>221</width>
      <height>32</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_8">
     <item>
      <widget class="QLabel" name="label_background_color">
       <property name="text">
        <string>Background color</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="background_combo">
       <item>
        <property name="text">
         <string>black</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>green</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>pink</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>purple</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>white</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>blue</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLineEdit" name="x_move_box">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>40</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="y_move_box">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>80</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="z_move_box">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>120</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="scale_box">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>330</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="x_rotation_box">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>200</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="y_rotation_box">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>240</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="z_rotation_box">
    <property name="geometry">
     <rect>
      <x>1120</x>
      <y>280</y>
      <width>41</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="qty_vertice">
    <property name="geometry">
     <rect>
      <x>110</x>
      <y>930</y>
      <width>131</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="qty_facet">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>930</y>
      <width>131</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qlview</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">qlview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
